models/llama3/reference_impl/generation.py

```diff
@@ -107,8 +112,10 @@ def build(
                 model_parallel_size = int(os.environ.get("WORLD_SIZE", 1))
             initialize_model_parallel(model_parallel_size)
 
-        local_rank = int(os.environ.get("LOCAL_RANK", 0))
-        torch.cuda.set_device(local_rank)
+        
```

Comment:
Execution will fail on line 108 above on non-cuda devices since "nccl" distributed backend is CUDA specific.